client/me/purchases/manage-purchase/index.jsx

```diff
@@ -272,6 +329,20 @@ class ManagePurchase extends Component {
 			}
 		}
 
+		if ( this.shouldShowNonPrimaryDomainWarning() ) {
+			const onClick = event => {
+				event.preventDefault();
+				trackNavItemClick( text );
+				this.showNonPrimaryDomainWarningDialog( link );
+			};
+
+			return (
```

Comment:
This duplicate definition of the `CompactCard` feels unnecessary.

An option would be to restructure this as:
```
let onClick = () => { trackNavItemClick( text ); }; // Feel free to reformat -- I still getting used to the new function notation. 😁 
if ( this.shouldShowNonPrimaryDomainWarning() ) {
	onClick = event => {
		event.preventDefault();
		trackNavItemClick( text );
		this.showNonPrimaryDomainWarningDialog( link );
	};
}
return (
	<CompactCard href={ link } onClick={ onClick }>
		{ text }
	</CompactCard>
);
```

Reply:
Good idea. Turns out we could simplify it even further :)